pkg/repository/git.go

```diff
@@ -64,6 +64,25 @@ func (r gitRepoController) calculateIsClean(ctx context.Context) (bool, error) {
 	return status.IsClean(), nil
 }
 
+func (r gitRepoController) calculateIsCleanBuildContext(ctx context.Context, buildContext string) (bool, error) {
```

Comment:
Can we reduce the naming of the function? I don't think the build context should be referenced in the name/param of the function. 

I'd name it `calculateIsCleanDir` or something similar. That way is more reusable

Reply:
cf69c7a86197f9bcf2929d9ca9aa1940200334ea